activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -491,7 +491,7 @@ def class_for_adapter(adapter)
 
         def each_current_configuration(environment, spec_name = nil)
           environments = [environment]
-          environments << "test" if environment == "development"
+          environments << "test" if environment == "development" && ENV["DATABASE_URL"]
```

Comment:
This is going to break apps that don't use database url.

It's saying "include test if the environment if development and database url is set". You want "include test if the environment is development and database url is not set".

It's not accidental that we include test in development mode when running this. It has been discussed extensively here https://github.com/rails/rails/issues/32926.

Reply:
Ah, yes, of course, that's what I meant to implement. I'll fix.